Client/src/ClientSceneManager/ClientSceneManager.hpp

```diff
@@ -0,0 +1,88 @@
+/*
+** EPITECH PROJECT, 2024
+** R-Type
+** File description:
+** ClientSceneManager
+*/
+
+#pragma once
+
+#include <unordered_map>
+
+#include <jsoncpp/json/json.h>
+
+#include "ISystem.hpp"
+#include "StringKeyMap.hpp"
+
+#define DEFAULT_SCENE "default.json"
+#define SCENE_PATH "Client/config/scenes/"
+#define LIB_COMPONENTS_PATH "Client/lib/components/"
+#define LIB_SYSTEMS_PATH "Client/lib/systems/"
+
+#define CONFIG_SUFFIX ".json"
+#define LIB_SUFFIX ".so"
+
+/**
+ * @brief Namespace for the scene manager
+ * This namespace contains all the classes or enums that are related to the scene management.
+ */
+namespace SceneManager {
+
+    /**
+     * @brief Client ClientSceneManager class
+     * It manages the scenes of the game and the transitions between them
+     */
+    class ClientSceneManager {
+
+        public:
+
+            /**
+             * @brief Construct a new ClientSceneManager object
+             * @param registries Registries for each scene
+             *
+             */
+            ClientSceneManager(std::shared_ptr<std::vector<ECS::Registry>> registries);
+
+            /**
+             * @brief Destroy the ClientSceneManager object
```

Comment:
```suggestion
             * @brief Destroy the ClientSceneManager object.
```
It's the same for all comments in this file.

Reply:
Done